core/src/test/java/io/parsingdata/metal/util/ParameterizedParse.java

```diff
@@ -44,7 +47,13 @@
 
     @Test
     public void test() throws IOException {
-        assertEquals(result, token.parse(env(parseState, encoding)).isPresent());
+        Optional<ParseState> endState = token.parse(env(parseState, encoding));
+        assertEquals(result, endState.isPresent());
+
+        endState.ifPresent(state -> {
```

Comment:
The `ifPresent()`-check is unnecessary I think, because the `assertEquals()` above it will terminate execution if it's not. So rewrite this to an `assertFalse()` over the `state.slice(ONE).isPresent()`-expression?

Reply:
I'm fine with style, else we'd need a `final ParseState state = endState.get()`.

Reply:
Oh, it's used to avoid a direct `get()`? I would just use `get()` if you do a hard check for `isPresent()` there anyway... But this is not a big thing to me.

Reply:
I prefer this style. It feels more robust to me, although the difference is small. When you use `get()` then the second test _must_ precede this tests, so there is a dependency. In this style, the order doesn't matter, the first test can even be removed without changing the behaviour of this test.

Reply:
I agree there are advantages in _defensiveness_ :-)

Reply:
It is a test class, so I'm fine with either :) But neither of the tests can be removed right? Removing the isPresent check will make everything succeed if nothing is present?